contracts/Comptroller.sol

```diff
@@ -927,6 +939,44 @@ contract Comptroller is ComptrollerV2Storage, ComptrollerInterface, ComptrollerE
         return (Error.NO_ERROR, maxBorrowOrRedeemAmount);
     }
 
+    /**
+      * @notice Set the given Asset caps for the given cToken markets. AssetCapSupply that brings total supply to or above asset cap will revert.
+      * @dev Admin or AssetCapGuardian function to set the Asset caps. A Asset cap of 0 corresponds to unlimited supply.
+      * @param cTokens The addresses of the markets (tokens) to change the supply caps for
+      * @param newAssetCaps The new borrow cap values in underlying to be set. A value of 0 corresponds to unlimited supply.
+      */
+
+        function _setMarketAssetCaps(CToken[] calldata cTokens, uint[] calldata newAssetCaps) external {
+    	require(msg.sender == admin || msg.sender == AssetCapGuardian, "only admin or borrow cap guardian can set borrow caps"); 
+
+        uint numMarkets = cTokens.length;
+        uint numAssetCaps = newAssetCaps.length;
+
+        require(numMarkets != 0 && numMarkets == numAssetCaps, "invalid input");
+
+        for(uint i = 0; i < numMarkets; i++) {
+            assetCaps[address(cTokens[i])] = newAssetCaps[i];
+            emit NewAssetCap(cTokens[i], newAssetCaps[i]);
+        }
+    }
+
+    /**
+     * @notice Admin function to change the Asset Cap Guardian
+     * @param newAssetCapGuardian The address of the new Asset Cap Guardian
+     */
+    function _setAssetCapGuardian(address newAssetCapGuardian) external {
+        require(msg.sender == admin, "only admin can set asset cap guardian");
```

Comment:
The use of require() does not conform to Compound's error handling system

Reply:
made Changes @TransmissionsDev